Added a user's activity to their profile page

aylinev2 opened the pull request:
- I added an area on a user's profile page that shows all the messages that the user has sent

aylinev2 committed:
added a user's activity to their profile page

acolwell reviewed (commented):
This looks like it would work, but I have some questions to help guide you toward something a little better. :)

1. What is the worst case runtime complexity for these changes? How could you improve on that?
2. What patterns do other parts of the code base use for filtering messages?
3. How would you refactor this code to make it more testable?

aylinev2 committed:
made fixes to way that a specific user's messages are retreived

aylinev2 commented:
- I added a hashmap that contains a list of a specific user's messages
- When the server is started up the setMessages() function is called so I thought it would be best to just sort the messages by author once right when the new list of messages is set
- I also added some code underneath addMessage() so that the sortMessagesByAuthor() function doesn't have to be called again each time theres a new message in the session

acolwell commented on src/main/java/codeu/model/store/basic/MessageStore.java:
```diff
@@ -70,6 +75,23 @@ private MessageStore(PersistentStorageAgent persistentStorageAgent) {
   }
 
+  /** Maps each message to its author. */
+  private void sortMessagesByAuthor(){
```
"sort" -> "split" or "sort" -> "filter" ? The main reason I'm suggesting this is "sort" implies to me that it is taking an message list and creating another list that is sorted by author. That isn't quite what you are doing here. Another option might be renaming this to something like "populateUserMessagesMap".

acolwell commented on src/main/java/codeu/model/store/basic/MessageStore.java:
```diff
@@ -70,6 +75,23 @@ private MessageStore(PersistentStorageAgent persistentStorageAgent) {
 
+  /** Maps each message to its author. */
+  private void sortMessagesByAuthor(){
+    for (Message message : messages){
```
Replace userMessages with an empty map so this doesn't end up containing old messages if setMessages() is called more than once.

acolwell commented on src/main/webapp/WEB-INF/view/profile.jsp:
```diff
@@ -48,6 +71,24 @@ TextProcessor processor = BBProcessorFactory.getInstance().createFromResource("k
+      <ul>
+     <%
+      for (Message message : messages) {
+        String author = UserStore.getInstance().getUser(message.getAuthorId()).getName();
```
I think you can avoid this lookup. Isn't message.getAuthorId() always equivalent to user.getId()? If so, I think you could just use user.getName() here instead.

Also consider moving the messages declaration to right before the loop. I think that would make it easier to see that the variable contains only messages for user.getId() and would add extra confidence that user.getName() is appropriate here.

acolwell reviewed (approved):
Looking pretty good to me. Just a few comments.

Also I believe if you add  "Fixes Issue #19 " to the merge message this pull request will automagically get associated with the issue.

aylinev2 committed:
small fixes to user activity portion
